extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/VertxHttpRecorder.java

```diff
@@ -547,7 +549,19 @@ public Verticle get() {
             @Override
             public void handle(AsyncResult<String> event) {
                 if (event.failed()) {
-                    futureResult.completeExceptionally(event.cause());
+                    Throwable effectiveCause = event.cause();
+                    if (effectiveCause instanceof BindException) {
+                        if ((sslConfig == null) && (httpServerOptions != null)) {
+                            effectiveCause = new QuarkusBindException(httpServerOptions.getPort());
```

Comment:
How would that work for negative ports (negative ports in Vert. x are random ports but all the verticles use the same port, unlike the port 0 which is random every time). I guess it's fine as it would indicate the negative port (like -1) and it's up to the user to check in the config which one it is.

Reply:
Yeah, although this is an edge case which I am going to guess we won't see often, if ever :)
I think we would probably just handle it when printing the helper messages if the exception fails.